packages/react/src/components/button.tsx

```diff
@@ -43,136 +49,120 @@ export type UploadButtonProps<TRouter extends FileRouter> =
-            const files = Array.from(e.target.files);
-            void startUpload(files, input);
-        },
-        disabled: !ready,
-    })
+  const getInputProps = () => ({
+    className: "hidden",
+    type: "file",
+    ref: fileInputRef,
+    multiple,
+    accept: generateMimeTypes(fileTypes ?? [])?.join(", "),
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+      if (!e.target.files) return;
+      const input = "input" in $props ? $props.input : undefined;
+      const files = Array.from(e.target.files);
+      void startUpload(files, input);
+    },
+    disabled: !ready,
+  });
 
-    const styleFieldArg: ButtonStyleFieldCallbackArgs = {
-        ready,
-        isUploading,
-        uploadProgress,
-        fileTypes,
-    };
+  const styleFieldArg: ButtonStyleFieldCallbackArgs = {
+    ready,
+    isUploading,
+    uploadProgress,
+    fileTypes,
+  };
 
-    const getUtState = () => {
-        if (!ready) return 'readying';
-        if (ready && !isUploading) return 'ready';
+  const state = (() => {
+    if (!ready) return "readying";
+    if (ready && !isUploading) return "ready";
 
-        return 'uploading';
-    }
+    return "uploading";
+  })();
 
-    return (
-        <div
-            className={twMerge(
-                "flex flex-col items-center justify-center gap-1",
-                $props.className,
-                styleFieldToClassName($props.appearance?.container, styleFieldArg),
-            )}
-            style={styleFieldToCssObject($props.appearance?.container, styleFieldArg)}
-            data-state={getUtState()}
-        >
-            <label
-                className={twMerge(
-                    classNames(
-                        "relative flex h-10 w-36 cursor-pointer items-center justify-center overflow-hidden rounded-md text-white after:transition-[width] after:duration-500",
-                        !ready && "cursor-not-allowed bg-blue-400",
-                        ready &&
-                        isUploading &&
-                        `bg-blue-400 after:absolute after:left-0 after:h-full after:bg-blue-600 ${progressHeights[uploadProgress]}`,
-                        ready && !isUploading && "bg-blue-600",
-                    ),
-                    styleFieldToClassName($props.appearance?.button, styleFieldArg),
-                )}
-                style={styleFieldToCssObject($props.appearance?.button, styleFieldArg)}
-                data-state={getUtState()}
-                data-ut-element="button"
-            >
-                <input
-                    {...getInputProps()}
-                />
-                {isUploading
-                    ? spinnerFieldToElement(
-                        $props.appearance?.buttonSpinner,
-                        styleFieldArg,
-                    ) || <Spinner />
-                    : contentFieldToContent($props.content?.button, styleFieldArg) ||
-                    getUploadButtonText(fileTypes)}
```

Comment:
i simplified this and removed the `buttonSpinner` prop. I don't think it makes sense to still control the content even if the user has overridden it.

Reply:
Fair enough, thank you